modules/network/src/main/java/org/apache/ignite/internal/network/scalecube/ScaleCubeTopologyService.java

```diff
@@ -191,8 +213,26 @@ public ClusterNode getByAddress(NetworkAddress addr) {
     /** {@inheritDoc} */
     @Override
     public @Nullable ClusterNode getByConsistentId(String consistentId) {
+        ClusterNode nodeInLogicalTopology = membersByConsistentIdInLogicalTopology.get(consistentId);
+        if (nodeInLogicalTopology != null) {
+            // Node is in the logical topology, check if it's in the physical topology. This could happen when topology is restored on
```

Comment:
It's not clear what 'could happen'. That the node is in the LT, but not in the PT? Anyway, please rephrase.